src/meltano/core/state_store/__init__.py

```diff
@@ -0,0 +1,121 @@
+"""State backends."""
+from __future__ import annotations
+
+import platform
+from enum import Enum
+from urllib.parse import urlparse
+
+from sqlalchemy.orm import Session
+
+from meltano.core.db import project_engine
+from meltano.core.project_settings_service import ProjectSettingsService
+from meltano.core.state_store.azure import AZStorageStateStoreManager
+from meltano.core.state_store.base import StateStoreManager
+from meltano.core.state_store.db import DBStateStoreManager
+from meltano.core.state_store.filesystem import (
+    LocalFilesystemStateStoreManager,
+    WindowsFilesystemStateStoreManager,
+)
+from meltano.core.state_store.google import GCSStateStoreManager
+from meltano.core.state_store.s3 import S3StateStoreManager
+
+
+class StateBackend(str, Enum):
+    """State backend."""
+
+    SYSTEMDB = "systemdb"
+    # These strings should match the expected scheme for URIs of
+    # the given type. E.g., filesystem state backends have a
+    # file://<path>/<to>/<state directory> URI
+    LOCAL_FILESYSTEM = "file"
+    AZURE = "azure"
+    S3 = "s3"
+    GCS = "gs"
+
+    @classmethod
+    def backends(cls) -> list[StateBackend]:
+        """List available state backends.
+
+        Returns:
+            List of available state backends.
+        """
+        return list(cls)
+
+    @property
+    def _managers(
+        self,
+    ) -> dict[
+        str,
+        type[DBStateStoreManager]
+        | type[LocalFilesystemStateStoreManager]
+        | type[WindowsFilesystemStateStoreManager]
+        | type[AZStorageStateStoreManager]
+        | type[GCSStateStoreManager]
+        | type[S3StateStoreManager],
```

Comment:
Would it make sense to use a type alias or some type of generic that references the superclass `StateStoreManager` here?

Not a big deal, though.

Reply:
Yeah I know this is pretty clunky. I initially had a type alias for this but since it required using the `type[]` syntax outside of an annotation (i.e. on the right hand side of an `=` rather than the right hand side of a `:`) it wouldn't work in <3.9.

What would using a Generic look like here?

Reply:
@cjohnhanson maybe generics are not even required and something like

```python
    @property
    def _managers(self) -> dict[str, type[StateStoreManager]:
        ...
```

might work because all the unioned types are derived from `StateStoreManager`. Wdyt?

Reply:
@edgarrmondragon yeah I initially had that but it fails mypy. It looks like we can use the generic `Mapping[str, type[StateStoreManager]]` though. I've updated this to use that instead,